File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        def bs( nums: List[int], target: int) -> int:
            l,r  = 0, len(nums)
            while l<r:
                m = (l+r)//2
                if nums[m] < target:
                    l = m  + 1
                else:
                    r = m
            return l
        
        l = bs(nums,target)
        if(l == len(nums) or nums[l] != target):
            return [-1,-1]
        r = bs(nums,target+1)
        return [l,r-1];
        

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int first = BinarySearch(nums, target);
        
        if(first == nums.size() || nums[first] != target) return {-1,-1};
        
        int last = BinarySearch(nums,target + 1);
        
        return {first,last-1};
    }
    
    int BinarySearch(vector<int>&nums, int target){
        int l = 0;
        int r = nums.size();
        
        while(l < r){
            int m = (l+r)/2;
            if(nums[m] < target) l = m + 1;
            else{
                r = m;
            }
        }
        return l;
    }
};